internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -729,6 +730,22 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, log logrus.FieldLogge
 	return reconcileComplete{dirty: true, stop: true}
 }
 
+func (r *BMACReconciler) isNonePlatform(ctx context.Context, cd *hivev1.ClusterDeployment) (isNone bool, err error) {
+	if cd.Spec.ClusterInstallRef == nil {
+		return false, errors.Errorf("Cluster Install Reference is null for cluster deployment ns=%s name=%s", cd.Namespace, cd.Name)
+	}
+	clusterInstall := hiveext.AgentClusterInstall{}
+	namespacedName := types.NamespacedName{
+		Namespace: cd.Namespace,
+		Name:      cd.Spec.ClusterInstallRef.Name,
+	}
+	if err = r.Get(ctx, namespacedName, &clusterInstall); err != nil {
+		return false,
+			errors.Wrapf(err, "AgentClusterInstall %s does not exist for ClusterDeployment %s", cd.Spec.ClusterInstallRef.Name, cd.Name)
```

Comment:
What do you think about propagating `err` as is rather than wrapping it under that message? I fear that this code may be saying that the ACI doesn't exist when the error could be something else. Another option would be to rephrase the error: 

```suggestion
			errors.Wrapf(err, "Could not get AgentClusterInstall %s for ClusterDeployment %s", cd.Spec.ClusterInstallRef.Name, cd.Name)
```

Reply:
OK. Will rephrase